prometheus/vec.go

```diff
@@ -25,10 +25,27 @@ import (
 // provided in this package.
 type MetricVec struct {
 	mtx      sync.RWMutex // Protects the children.
-	children map[uint64]Metric
+	children map[uint64][]metricLabelValue
 	desc     *Desc
 
 	newMetric func(labelValues ...string) Metric
+	hashAdd   func(h uint64, s string) uint64 // replace hash function for testing collision handling
+}
+
+// newMetricVec returns an initialized MetricVec. The concrete value is
+// returned for embedding into another struct.
+func newMetricVec(desc *Desc, newMetric func(lvs ...string) Metric) MetricVec {
+	return MetricVec{
+		children:  map[uint64][]metricLabelValue{},
+		desc:      desc,
+		newMetric: newMetric,
+		hashAdd:   hashAdd,
+	}
+}
+
+type metricLabelValue struct {
```

Comment:
Perhaps call this `metricWithLabelValues`? When reading the code where this type is used, `metricLabelValue` suggest it's a single label value of a metric.